lib/dialyzer/test/small_SUITE_data/src/guards.erl

```diff
@@ -145,6 +145,14 @@ t17(X) when #{x => X} ->
 t18(X) when <<X>> ->
     ok.
 
+%% Should give a warning, see https://github.com/erlang/otp/issues/7181
+t19() when is_function(ok, -9223372036854775808) ->
+    never.
+
+% Could give a warning, see the same link
+t20(X) ->
+    (is_function(ok, 9223372036854775807) andalso X) bor X.
```

Comment:
The following warnings will be generated:

```
guards.erl:153:1: Function t20/1 has no local return
guards.erl:154:22: The call erlang:is_function
         ('ok',
          9223372036854775807) will never return since it differs in the 2nd argument from the success typing arguments: 
         (any(),
          byte())
```

This is not correct, because this call to `is_function/2` will not fail but return `false`:

```
1> is_function(ok, 9223372036854775807).
false
```

A negative arity will cause the call to fail:

```
2> is_function(ok, -1).
** exception error: bad argument
     in function  is_function/2
        called as is_function(ok,-1)
        *** argument 2: out of range
```